### PR TITLE
Allow to set options for barman connectivity

### PR DESCRIPTION
**Pull request by chtitux**
- Allow to pass arbitrary options
- Build the URL in a dedicated template
- Allow to specify path prefix for barman files
- Add documentation in [README.md](README.md)

### CODE REVIEW

File: tasks/postgres-standby-barman.yml [hunk 1]
```diff
@@ -1,4 +1,8 @@
 ---
+- name: Compute barman remote URL
+  set_fact:
+    postgres_barman_remote_url: "{{ lookup('template', 'barman_remote_url.j2') | trim }}"
```
Comment (paulrbr-fl) [inline tasks/postgres-standby-barman.yml]: I feel that 'templates' in an Ansible role should contain mostly configuration template files. This template is here only to help construct this `fact`.

What do you think about using ansible's `vars:` instead?

Something like:

```
  set_fact:
    postgres_barman_remote_url: "{{ scheme }}{{ user }}{{ server }}{{ delimiter }}{{ path }}"
  vars:
    scheme: {{ postgres_barman_rsync_enabled | ternary('rsync://', '') }}
    user: {{ postgres_barman_remote_user is defined | ternary(postgres_barman_remote_user + '@', '') }}
    server: {{ postgres_barman_server }}
    delimiter: {{ not postgres_barman_rsync_enabled | ternary(':', '')}}
    path: {{ postgres_barman_path_prefix }}
```

I think it helps readability. WDYT?

File: templates/standby-clone.sh.j2 [hunk 2]
```diff
@@ -29,11 +29,8 @@ sudo -u postgres mkdir -p /var/lib/postgresql/{{ postgres_version }}/{{ postgres
 
 echo Get previous backup from backups server
 sudo -u postgres \
-  time rsync --progress -pvia --exclude='*.conf' --exclude='server.crt' --exclude='server.key' --delete \
-{% if postgres_barman_rsync_enabled|default(false) -%}
-  --password-file=/var/lib/postgresql/.rsync_pass \
```
Comment (paulrbr-fl) [inline templates/standby-clone.sh.j2:34]: what happens to this default value that disappears from your change?

**Review by paulrbr-fl — COMMENTED** [review]
Looks good overall. I left a comment about the style of building variables. WDYT?

File: tasks/postgres-standby-barman.yml (hunk 1 above)
Comment (chtitux) [reply tasks/postgres-standby-barman.yml]: It definitely helps, thanks a lot @paulRbr !

I have slightly adapted your patch so we don't use `not … | ternary(…)` but `… | ternary(…)` and added the password file path option by default when using rsync, as we create it in the role.

File: templates/standby-clone.sh.j2 (hunk 2 above)
Comment (chtitux) [reply templates/standby-clone.sh.j2:34]: I have re-added it in the `set_fact` step (if using rsync)

**Comment by paulrbr-fl** [comment]
purfect! That's much more readable! Thanks @chtitux

**Pull request merged**